### ISSUE TITLE
Date field hides date when trying to edit existing date

### ISSUE THREAD

**Issue by Chris-Petty** [openmsupply/mobile#517]
Build Number: 2.0.0

Description: 
Date field hides date when trying to edit existing date

Reproducible: Yes

Reproduction Steps:
1. Go to Supplier Invoices
2. Create a new supplier invoice 
3. Add an item and add quantity + date
4. Change focus away from date cell
5. Select date cell again - The existing date is hidden

Comments:

**Comment by GaryWilletts** [comment]
Which date field would that be? Or is there only one?

**Comment by Chris-Petty** [comment]
It happens anywhere it's used... Which is only in a new supplier invoice made on mobile at this point.

**Comment by andreievg** [comment]
This is because of the way that 'masking' was implemented in ExpiryTextInput, and the masking needed to be used because '/' is not available in all number keyboard (noticed that when we first tried the new app on Tab A's).  
So the ExpiryTextInput puts a front slash when it recognises one is needed, appends 0 to month if first digit entered is > 1, it also appends required amount of digits to year field. This TextInput does not allow illegal dates to be entered at all, so if the previous date is the default value for this input, and it needs to be changed, then it needs to be fully deleted (i.e. can't set the cursor on a month and delete it, have to re-enter the expiry date from the start). This is why the Input clears previous expiry date.
By the way, if you click out of the ExpiryTextInput or focus out of it with incomplete expiry date, it will revert back to the previous set date.
